src/spanned.rs

```diff
@@ -110,6 +110,18 @@ mod private {
     impl<T: ToTokens> Sealed for T {}
 }
 
+// FIXME: This shouldn't be required, since optimally
+// spans should never be invalid. This can probably be removed when
+// https://github.com/rust-lang/rust/issues/43081 is resolved.
+fn check_invalid_spans(span: Span) -> Span {
+    let debug = format!("{:?}", span);
+    if &debug[debug.len() - 11..] == "bytes(0..0)" {
```

Comment:
I would prefer to use https://doc.rust-lang.org/std/primitive.str.html#method.ends_with to avoid a panic in the case that the debug format for Span changes to a shorter format.

Reply:
That is a very good point.